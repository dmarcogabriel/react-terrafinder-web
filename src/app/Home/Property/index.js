import React from 'react';
import Button from 'common/Button';
import Input from 'common/Input';
import { MdArrowForward } from 'react-icons/md';
import classes from './Property.module.scss';
import propertyImg from './static/background-banner1.jpg';
import ownerPlaceholderImg from './static/ownerPlaceholder.png';
import TextArea from './TextArea';

export default function Property() {
  return (
    <div className={classes.container}>
      <h1>Fazenda Campo Alto - Três Lagoas</h1>

      <div className={classes.content}>
        <div className={classes.propertySession}>
          <div className={classes.propertyImage}>
            <img src={propertyImg} alt="" />
          </div>

          <div className={classes.col}>
            <div className={classes.row}>
              <div className={classes.session}>
                <p className={classes.title}>Resumo</p>

                <p>Valor: R$ 11.000,00</p>
                <p>Área: 300ha</p>
                <p>Código: #1020A</p>
                <p>Cidade mais próxima: Três Lagoas</p>
                <p>Estado: Mato Grosso do Sul</p>
                <p>Tipo de Propriedade: Fazenda</p>
                <p>Cultivo: Soja / Milho</p>
              </div>

              <div className={classes.session}>
                <p className={classes.title}>Atividades e Destaques</p>

                <ul>
                  <li>Casa com 2 quartos</li>
                  <li>Caseiro</li>
                  <li>Animais domésticos</li>
                  <li>Sinal de rede wifi</li>
                  <li>Temporada de caça legalizada</li>
                  <li>Fácil localização</li>
                  <li>Cultivos alternativos</li>
                  <li>Registrado</li>
                </ul>
              </div>
            </div>

            <div>
              <div className={classes.session}>
                <p className={classes.title}>Descrição da Propriedade</p>

                <p>
                  Lorem ipsum dolor sit amet consectetur adipisicing elit.
                  Eveniet consectetur sed laudantium excepturi vero earum,
                  nostrum autem ut natus ad accusantium nobis aperiam reiciendis
                  quam deserunt. Neque quia iste mollitia!
                </p>
              </div>
            </div>
          </div>
        </div>

        <div className={classes.ownerSession}>
          <div className={classes.ownerAvatar}>
            <img src={ownerPlaceholderImg} alt="" />
          </div>

          <p className={classes.ownerName}>Caetano Borges</p>

          <p>Entre em contato com o proprietário pelo formulário abaixo:</p>

          <Input label="Nome" placeholder="João da Silva" />
          <Input label="E-mail" placeholder="joao@dasilva.com" />
          <Input label="Telefone" placeholder="(11) 9999-9999" />

          <TextArea label="Escreva aqui sua mensagem para o proprietário" />

          <Button className={classes.sendMessageButton}>
            <p>Enviar mensagem ao proprietário</p>

            <MdArrowForward size={22} />
          </Button>
        </div>
      </div>
    </div>
  );
}
