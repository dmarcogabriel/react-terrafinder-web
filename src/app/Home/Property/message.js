export const message = (ownerName, propertyName) => {
  return `Ol√°, ${ownerName}! Estou interessado na sua propriedade chamada ${propertyName}!
  Qual a melhor forma de pagamento?
  `;
};
