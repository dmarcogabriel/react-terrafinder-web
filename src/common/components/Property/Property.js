import React from 'react';
import { moneyFormat, farmingFormat } from 'utils/formatters';
import imagePlaceholder from 'common/static/soja.jpg';
import { Card, CardMedia, Box, Typography } from '@mui/material';
import {
  FullscreenOutlined as SizeIcon,
  Map as MapIcon,
  Spa as FarmingIcon,
  Star as StarIcon,
  Edit as EditIcon,
} from '@mui/icons-material';
import {
  PropertyAmount,
  PropertyAtributesBox,
  PropertyAttribute,
  PropertyPremiumLabel,
  PropertyEditButton,
} from './styles';

export const Property = ({
  dataTestId,
  property,
  onSelect,
  isPremium,
  isEditable,
  onClickEdit,
  containerSx = {},
}) => {
  const [photo] = property.photos;

  const handleSelect = () => onSelect && onSelect(property._id);

  const handleEdit = () => onClickEdit && onClickEdit(property._id);

  return (
    <Card
      data-testid={dataTestId || 'property'}
      sx={{ my: 1, cursor: 'pointer', position: 'relative', ...containerSx }}
    >
      {isPremium && (
        <PropertyPremiumLabel sx={{ p: 1 }}>
          <StarIcon fontSize="small" sx={{ mr: 1 }} />
          An√∫ncio Premium
        </PropertyPremiumLabel>
      )}
      {isEditable && (
        <PropertyEditButton
          variant="contained"
          color="secondary"
          onClick={handleEdit}
        >
          <EditIcon fontSize="large" />
        </PropertyEditButton>
      )}
      <CardMedia
        image={
          photo
            ? `${process.env.REACT_APP_STATIC}/images/${photo}`
            : imagePlaceholder
        }
        alt={property.name}
        sx={{ height: 200 }}
      />
      <Box
        sx={{
          py: 2,
          pl: 2,
          display: 'flex',
          alignItems: 'center',
          justifyContent: 'space-between',
        }}
        onClick={handleSelect}
      >
        <Typography variant="h6">{property.name}</Typography>
        <PropertyAmount sx={{ p: 1 }}>
          {moneyFormat(property.amount, false)}
        </PropertyAmount>
      </Box>
      <PropertyAtributesBox sx={{ py: 2, px: 2 }}>
        <PropertyAttribute sx={{ my: { xs: 1, md: 0 } }}>
          <SizeIcon color="success" />
          <Typography sx={{ ml: 1 }}>{property.size}ha</Typography>
        </PropertyAttribute>
        <PropertyAttribute sx={{ my: { xs: 1, md: 0 } }}>
          <FarmingIcon color="success" />
          <Typography sx={{ ml: 1 }}>
            {farmingFormat(property.farming)}
          </Typography>
        </PropertyAttribute>
        <PropertyAttribute sx={{ my: { xs: 1, md: 0 } }}>
          <MapIcon color="success" />
          <Typography sx={{ ml: 1 }}>{property.state}</Typography>
        </PropertyAttribute>
      </PropertyAtributesBox>
    </Card>
  );
};
