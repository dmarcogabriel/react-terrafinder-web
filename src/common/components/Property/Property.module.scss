@import 'styles';

.propertyItem {
  box-shadow: $shadow-base;
  margin: 1.25rem 0;
  cursor: pointer;

  > img {
    width: 100%;
  }

  .content {
    position: relative;
    padding: 1rem;

    .name {
      font-size: 1.25rem;
      margin: 0.5rem 0;
    }

    .amount {
      display: inline-block;
      padding: 0.5rem;
      background: $primary-dark;
      color: $white;
      border-radius: 0.2rem;
      margin: 0.5rem 0;
      font-size: 1.25rem;

      @media ($desktop) {
        position: absolute;
        top: 1rem;
        right: 0;
        border-bottom-right-radius: 0;
        border-top-right-radius: 0;
      }
    }

    .attributes {
      @media ($desktop) {
        display: flex;
        justify-content: space-between;
        width: 70%;
      }

      .attribute {
        display: flex;
        align-items: center;
        margin: 0.5rem 0;

        > p {
          margin-left: 1rem;
          font-weight: bold;
        }

        .icon {
          color: $green;
        }
      }
    }
  }
}
