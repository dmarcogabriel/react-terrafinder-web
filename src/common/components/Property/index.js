import React from 'react';
import { AiOutlineFullscreen } from 'react-icons/ai';
import { MdMap } from 'react-icons/md';
import { RiPlantFill } from 'react-icons/ri';
import { moneyFormat } from 'utils/formatters';
import classes from './Property.module.scss';

export default function Property({ property, index, onSelect }) {
  const [photo] = property.photos;

  return (
    <div
      role="button"
      tabIndex={index}
      onKeyDown={() => onSelect(property._id)}
      className={classes.propertyItem}
      onClick={() => onSelect(property._id)}
    >
      <img src={`http://localhost:8000/images/${photo}`} alt={photo} />

      <div className={classes.content}>
        <p className={classes.name}>{property.name}</p>

        <p className={classes.amount}>{moneyFormat(property.amount, false)}</p>

        <div className={classes.attributes}>
          <div className={classes.attribute}>
            <AiOutlineFullscreen size={22} className={classes.icon} />
            <p>{property.size}ha</p>
          </div>

          <div className={classes.attribute}>
            <RiPlantFill size={22} className={classes.icon} />
            <p>{property.farming[0]}</p>
          </div>

          <div className={classes.attribute}>
            <MdMap size={22} className={classes.icon} />
            <p>{property.state}</p>
          </div>
        </div>
      </div>
    </div>
  );
}
