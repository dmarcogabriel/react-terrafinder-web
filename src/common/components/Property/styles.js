import React from 'react';
import { styled } from '@mui/material/styles';
import { Typography, Box, Button, ButtonBase } from '@mui/material';

export const ClickableArea = styled(ButtonBase)({
  // styles here...
});

export const PropertyPremiumLabel = styled(Typography)({
  backgroundColor: '#ffcb08',
  borderBottomLeftRadius: '0.2rem',
  borderBottomRightRadius: '0.2rem',
  position: 'absolute',
  left: '1.5rem',
  display: 'flex',
  alignItems: 'center',
});

export const PropertyAmount = styled(({ children, ...props }) => (
  <Typography {...props}>{children}</Typography>
))(({ theme }) => ({
  backgroundColor: '#ffcb08',
  borderTopLeftRadius: '0.2rem',
  borderBottomLeftRadius: '0.2rem',

  [theme.breakpoints.up('md')]: {
    // position: absolute;
    // top: 1rem;
    // right: 0;
    // border-bottom-right-radius: 0;
    // border-top-right-radius: 0;
  },
}));

export const PropertyAtributesBox = styled(Box)(({ theme }) => ({
  [theme.breakpoints.up('md')]: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
}));

export const PropertyAttribute = styled(Box)({
  display: 'flex',
  alignItems: 'center',
});
