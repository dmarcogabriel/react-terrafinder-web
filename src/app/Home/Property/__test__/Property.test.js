import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
// import { render } from '@testing-library/react';
import Property from '..';

describe('<Property />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Property />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  // Add tests here...

  it('matches snapshot', () => {
    const tree = renderer.create(<Property />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
