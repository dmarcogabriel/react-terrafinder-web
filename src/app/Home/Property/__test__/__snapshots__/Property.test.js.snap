// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Property /> matches snapshot 1`] = `
<div>
  <h1>
    Fazenda Campo Alto - Três Lagoas
  </h1>
  <div>
    <div>
      <div>
        <img
          alt=""
          src="test-file-stub"
        />
      </div>
      <div>
        <div>
          <div>
            <p>
              Resumo
            </p>
            <p>
              Valor: R$ 11.000,00
            </p>
            <p>
              Área: 300ha
            </p>
            <p>
              Código: #1020A
            </p>
            <p>
              Cidade mais próxima: Três Lagoas
            </p>
            <p>
              Estado: Mato Grosso do Sul
            </p>
            <p>
              Tipo de Propriedade: Fazenda
            </p>
            <p>
              Cultivo: Soja / Milho
            </p>
          </div>
          <div>
            <p>
              Atividades e Destaques
            </p>
            <ul>
              <li>
                Casa com 2 quartos
              </li>
              <li>
                Caseiro
              </li>
              <li>
                Animais domésticos
              </li>
              <li>
                Sinal de rede wifi
              </li>
              <li>
                Temporada de caça legalizada
              </li>
              <li>
                Fácil localização
              </li>
              <li>
                Cultivos alternativos
              </li>
              <li>
                Registrado
              </li>
            </ul>
          </div>
        </div>
        <div>
          <div>
            <p>
              Descrição da Propriedade
            </p>
            <p>
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Eveniet consectetur sed laudantium excepturi vero earum, nostrum autem ut natus ad accusantium nobis aperiam reiciendis quam deserunt. Neque quia iste mollitia!
            </p>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div>
        <img
          alt=""
          src="test-file-stub"
        />
      </div>
      <p>
        Caetano Borges
      </p>
      <p>
        Entre em contato com o proprietário pelo formulário abaixo:
      </p>
      <div>
        <div>
          <p>
            Nome
          </p>
          <span>
            <svg
              fill="currentColor"
              height={22}
              size={22}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width={22}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12,2C6.486,2,2,6.486,2,12s4.486,10,10,10s10-4.486,10-10S17.514,2,12,2z M12,20c-4.411,0-8-3.589-8-8s3.589-8,8-8 s8,3.589,8,8S16.411,20,12,20z"
              />
              <path
                d="M14.829,14.828c-0.185,0.184-0.384,0.349-0.592,0.489c-0.217,0.146-0.445,0.27-0.68,0.369 c-0.244,0.103-0.496,0.181-0.749,0.233c-0.531,0.108-1.087,0.108-1.616,0c-0.254-0.052-0.506-0.13-0.75-0.233 c-0.234-0.099-0.463-0.223-0.679-0.369c-0.209-0.141-0.408-0.305-0.593-0.489c-0.181-0.181-0.346-0.38-0.488-0.592l-1.658,1.119 c0.215,0.318,0.462,0.617,0.734,0.889c0.273,0.273,0.572,0.52,0.887,0.731c0.323,0.218,0.666,0.404,1.02,0.553 c0.365,0.154,0.744,0.272,1.128,0.35C11.189,17.959,11.596,18,12,18s0.811-0.041,1.208-0.122c0.383-0.078,0.762-0.196,1.127-0.35 c0.354-0.149,0.696-0.335,1.021-0.553c0.313-0.212,0.612-0.458,0.886-0.731c0.272-0.271,0.52-0.571,0.734-0.889l-1.658-1.119 C15.175,14.448,15.01,14.647,14.829,14.828z"
              />
              <circle
                cx="8.5"
                cy="10.5"
                r="1.5"
              />
              <circle
                cx="15.493"
                cy="10.493"
                r="1.493"
              />
            </svg>
          </span>
        </div>
        <input
          onChange={[Function]}
          placeholder="João da Silva"
        />
      </div>
      <div>
        <div>
          <p>
            E-mail
          </p>
          <span>
            <svg
              fill="currentColor"
              height={22}
              size={22}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width={22}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12,2C6.486,2,2,6.486,2,12s4.486,10,10,10s10-4.486,10-10S17.514,2,12,2z M12,20c-4.411,0-8-3.589-8-8s3.589-8,8-8 s8,3.589,8,8S16.411,20,12,20z"
              />
              <path
                d="M14.829,14.828c-0.185,0.184-0.384,0.349-0.592,0.489c-0.217,0.146-0.445,0.27-0.68,0.369 c-0.244,0.103-0.496,0.181-0.749,0.233c-0.531,0.108-1.087,0.108-1.616,0c-0.254-0.052-0.506-0.13-0.75-0.233 c-0.234-0.099-0.463-0.223-0.679-0.369c-0.209-0.141-0.408-0.305-0.593-0.489c-0.181-0.181-0.346-0.38-0.488-0.592l-1.658,1.119 c0.215,0.318,0.462,0.617,0.734,0.889c0.273,0.273,0.572,0.52,0.887,0.731c0.323,0.218,0.666,0.404,1.02,0.553 c0.365,0.154,0.744,0.272,1.128,0.35C11.189,17.959,11.596,18,12,18s0.811-0.041,1.208-0.122c0.383-0.078,0.762-0.196,1.127-0.35 c0.354-0.149,0.696-0.335,1.021-0.553c0.313-0.212,0.612-0.458,0.886-0.731c0.272-0.271,0.52-0.571,0.734-0.889l-1.658-1.119 C15.175,14.448,15.01,14.647,14.829,14.828z"
              />
              <circle
                cx="8.5"
                cy="10.5"
                r="1.5"
              />
              <circle
                cx="15.493"
                cy="10.493"
                r="1.493"
              />
            </svg>
          </span>
        </div>
        <input
          onChange={[Function]}
          placeholder="joao@dasilva.com"
        />
      </div>
      <div>
        <div>
          <p>
            Telefone
          </p>
          <span>
            <svg
              fill="currentColor"
              height={22}
              size={22}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width={22}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12,2C6.486,2,2,6.486,2,12s4.486,10,10,10s10-4.486,10-10S17.514,2,12,2z M12,20c-4.411,0-8-3.589-8-8s3.589-8,8-8 s8,3.589,8,8S16.411,20,12,20z"
              />
              <path
                d="M14.829,14.828c-0.185,0.184-0.384,0.349-0.592,0.489c-0.217,0.146-0.445,0.27-0.68,0.369 c-0.244,0.103-0.496,0.181-0.749,0.233c-0.531,0.108-1.087,0.108-1.616,0c-0.254-0.052-0.506-0.13-0.75-0.233 c-0.234-0.099-0.463-0.223-0.679-0.369c-0.209-0.141-0.408-0.305-0.593-0.489c-0.181-0.181-0.346-0.38-0.488-0.592l-1.658,1.119 c0.215,0.318,0.462,0.617,0.734,0.889c0.273,0.273,0.572,0.52,0.887,0.731c0.323,0.218,0.666,0.404,1.02,0.553 c0.365,0.154,0.744,0.272,1.128,0.35C11.189,17.959,11.596,18,12,18s0.811-0.041,1.208-0.122c0.383-0.078,0.762-0.196,1.127-0.35 c0.354-0.149,0.696-0.335,1.021-0.553c0.313-0.212,0.612-0.458,0.886-0.731c0.272-0.271,0.52-0.571,0.734-0.889l-1.658-1.119 C15.175,14.448,15.01,14.647,14.829,14.828z"
              />
              <circle
                cx="8.5"
                cy="10.5"
                r="1.5"
              />
              <circle
                cx="15.493"
                cy="10.493"
                r="1.493"
              />
            </svg>
          </span>
        </div>
        <input
          onChange={[Function]}
          placeholder="(11) 9999-9999"
        />
      </div>
      <div>
        <div>
          <p>
            Escreva aqui sua mensagem para o proprietário
          </p>
          <span>
            <svg
              fill="currentColor"
              height={22}
              size={22}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width={22}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12,2C6.486,2,2,6.486,2,12s4.486,10,10,10s10-4.486,10-10S17.514,2,12,2z M12,20c-4.411,0-8-3.589-8-8s3.589-8,8-8 s8,3.589,8,8S16.411,20,12,20z"
              />
              <path
                d="M14.829,14.828c-0.185,0.184-0.384,0.349-0.592,0.489c-0.217,0.146-0.445,0.27-0.68,0.369 c-0.244,0.103-0.496,0.181-0.749,0.233c-0.531,0.108-1.087,0.108-1.616,0c-0.254-0.052-0.506-0.13-0.75-0.233 c-0.234-0.099-0.463-0.223-0.679-0.369c-0.209-0.141-0.408-0.305-0.593-0.489c-0.181-0.181-0.346-0.38-0.488-0.592l-1.658,1.119 c0.215,0.318,0.462,0.617,0.734,0.889c0.273,0.273,0.572,0.52,0.887,0.731c0.323,0.218,0.666,0.404,1.02,0.553 c0.365,0.154,0.744,0.272,1.128,0.35C11.189,17.959,11.596,18,12,18s0.811-0.041,1.208-0.122c0.383-0.078,0.762-0.196,1.127-0.35 c0.354-0.149,0.696-0.335,1.021-0.553c0.313-0.212,0.612-0.458,0.886-0.731c0.272-0.271,0.52-0.571,0.734-0.889l-1.658-1.119 C15.175,14.448,15.01,14.647,14.829,14.828z"
              />
              <circle
                cx="8.5"
                cy="10.5"
                r="1.5"
              />
              <circle
                cx="15.493"
                cy="10.493"
                r="1.493"
              />
            </svg>
          </span>
        </div>
        <textarea
          onChange={[Function]}
          rows={5}
        />
      </div>
      <button
        className="button"
        type="button"
      >
        <p>
          Enviar mensagem ao proprietário
        </p>
        <svg
          fill="currentColor"
          height={22}
          size={22}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width={22}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
